libmisc/find_new_gid.c

```diff
@@ -183,7 +189,7 @@ int find_new_gid (bool sys_group,
 			 * gr_locate_gid() found the GID in an as-yet uncommitted
 			 * entry. We'll proceed below and auto-set a GID.
 			 */
-		} else if (result == EEXIST || result == ERANGE) {
+		} else if (result == EEXIST || result == ERANGE || result == EINVAL) {
 			/*
 			 * Continue on below. At this time, we won't
 			 * treat these two cases differently.
```

Comment:
You should update this comment as you added a new case

Reply:
I rather went ahead and adjusted all of them. It looked like we might not need to differentiate between `EEXIST` and `EINVAL`, but I went for the safer route.